packages/git/src/browser/git-scm-provider.ts

```diff
@@ -260,17 +260,19 @@ export class GitScmProvider implements ScmProvider {
 
     async stageAll(): Promise<void> {
         try {
-            const { repository, unstagedChanges } = this;
-            const uris = unstagedChanges.map(c => c.uri);
-            await this.git.add(repository, uris);
+            // TODO resolve deletion conflicts
+            // TODO confirm staging unresolved files
+            await this.git.add(this.repository, []);
         } catch (error) {
             this.gitErrorHandler.handleError(error);
         }
     }
     async stage(uri: string): Promise<void> {
         try {
-            const { repository, unstagedChanges } = this;
-            if (unstagedChanges.some(change => change.uri === uri)) {
+            const { repository, stagedChanges } = this;
+            if (!stagedChanges.some(change => change.uri === uri)) {
```

Comment:
the way it was before should be good. this line breaks it. can you explain the intent, please?

![2019-06-14 07 31 35](https://user-images.githubusercontent.com/914497/59485580-aa8a8280-8e76-11e9-94d5-49cd6d04f121.gif)

I cannot stage this new change to a file which already was staged before.

Reply:
Maybe we should express the condition clearly; is this what you wanted to check for?
`const dirty = unstagedChanges.some(change => chan`ge.uri === uri) || mergeChanges.some(change => change.uri === uri);`

Reply:
just tested quickly; with the proposed change I could stage changes for already staged files and  it played well with the merge conflict changes.